### PR TITLE
dts: xtensa: esp32: modify esp32 dts

### PR DESCRIPTION
**Pull request by uLipe**
To split ipm area into ipm memory and
general use shared memory.

Signed-off-by: Felipe Neves <felipe.neves@linaro.org>

### CODE REVIEW

File: dts/xtensa/espressif/esp32.dtsi
Comment (carlocaione) [inline dts/xtensa/espressif/esp32.dtsi:126]: nit: can you leave it as hex?

**Review by dcpleung — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Comment by uLipe** [comment]
@cfriedt @stephanosio , is this could be merged as well before 3.3? It only touches in the esp32 located dts file and the change is very small but valuable to enabling further work by integrating it to openAMP samples in near future.

Thank you.

**Comment by stephanosio** [comment]
Since this is not fixing a bug, it will need to wait until merge window is open next week.

**Pull request merged**
